client.go

```diff
@@ -39,8 +41,9 @@ func Dial(addr string, opts ...ConnOption) (*Client, error) {
 }
 
 // New establishes an AMQP client connection over conn.
-func New(conn net.Conn, opts ...ConnOption) (*Client, error) {
-	c, err := newConn(conn, opts...)
+// opts: pass nil to accept the default values.
+func New(conn net.Conn, opts *ConnOptions) (*Client, error) {
```

Comment:
Unrelated, I think we should rename `Client` to `Connection` so it aligns with the other implementations.

Reply:
Agreed on this, if just to match what it truly is and be even more 1:1 with AMQP.